Lib/test/test_ssl.py

```diff
@@ -4815,24 +4843,28 @@ def call_after_accept(conn_to_client):
                 tls_client.close()
 
         server.join()
-        self.assertEqual(b"", received_data)
-        self.assertIsInstance(wrap_error, OSError)  # All platforms.
-        self.non_linux_skip_if_other_okay_error(wrap_error)
-        self.assertIsInstance(wrap_error, ssl.SSLError)
-        self.assertIn("before TLS handshake with data", wrap_error.args[1])
-        self.assertIn("before TLS handshake with data", wrap_error.reason)
-        self.assertNotEqual(0, wrap_error.args[0])
-        self.assertIsNone(wrap_error.library, msg="attr must exist")
+        try:
+            self.assertEqual(b"", received_data)
+            self.assertIsInstance(wrap_error, OSError)  # All platforms.
+            self.non_linux_skip_if_other_okay_error(wrap_error)
+            self.assertIsInstance(wrap_error, ssl.SSLError)
+            self.assertIn("before TLS handshake with data", wrap_error.args[1])
+            self.assertIn("before TLS handshake with data", wrap_error.reason)
+            self.assertNotEqual(0, wrap_error.args[0])
+            self.assertIsNone(wrap_error.library, msg="attr must exist")
+        finally:
+            # gh-108342: Explicitly break the reference cycle
+            wrap_error = None
+            server = None
 
     def test_https_client_non_tls_response_ignored(self):
-
         server_responding = threading.Event()
 
         class SynchronizedHTTPSConnection(http.client.HTTPSConnection):
             def connect(self):
-                http.client.HTTPConnection.connect(self)
+                super().connect()
```

Comment:
This was intentionally not calling super().  it called http.client.**HTTP**Connection's connect, skipping http.client.**HTTPS**Connection's connect() on purpose to skip up a level.  On purpose.

because **HTTPS**Connection connect calls _wrap_socket and that is what we needed to avoid because we do it ourselves below.

Leave the previous **HTTP**Connection connect call and just add an explanatory comment to make it clear.

Reply:
Oh, that wasn't obvious when I read the code! Ok, I will revert and add a comment to explain this subtle function call :-)

Reply:
yeah i don't know how i managed to not leave a comment about that. definitely unusual. :)